### PR TITLE
Diver + mathematician + logician visitors (count depth, numbers and operations)

### PR DESCRIPTION
**Pull request by Nephty**
Added a visitor that computes the depth of expressions.

Simple numbers have depth 0, an expression adds one to the depth.

Example : 

0 has depth 0
(1+1) has depth 1
((1+1)+1) has depth 1
((1+1)+((1+1)+1)) has depth 2

### CODE REVIEW

**Review by MrRose765 — DISMISSED** [review]
lgtm :+1:

**Comment by VionFrancois** [comment]
Nice feature mate

**Review by VionFrancois — APPROVED** [review]
Nice feature mate

**Comment by MrRose765** [comment]
- pls remove functions such as _countDepth()_ in Operation.java
- also you forgot to do the same for countOps and countNbs

**Comment by Nephty** [comment]
This branch was only supposed to add the visitor that handles the depth method but @LicorneRose765 tricked me into implementing the number counter and operations counter visitor as well (on this same branch). Sorry for the bad practice

**Comment by Nephty** [comment]
For the reviewers :

Check in TestCounting if we can just remove the whole test file because I remove tests of deprecated depth, countNbs and countOps methods

**Comment by Nephty** [comment]
Last commit was a somewhat important modification of the functionning of the visitor. You can find a comment in the commit https://github.com/Nephty/calculator-cucumber-2024/pull/6/commits/d1fed697837c8f6867ba13eaea02243bb93ea90c explaining the slight modification or read it below :

> moved visit(all args) from Operation abstract class into specific visitors
It seemed more appropriate to move this behavior into the visitors. It is the action of visiting an expression that should trigger the acceptance of each argument of this expression. Until now, it was the acceptance of a visitor that instantly triggered the acceptance of the visitor by each argument which was a skipping of the visiting action.
In summary : trigger the acceptance of children because of the visiting of the expression and not because of the acceptance of that expression. Acceptance only triggers the visiting of the expression

Note that it is not a huge modification in behavior, but rather a shift of the location of some logic.

**Review by MrRose765 — APPROVED** [review]
I think it's ok now.  Good job :+1: 
Concerning the file TestCounting, It seems that you can now delete it, since you did the same job in TestLogician.

**Review by VionFrancois — APPROVED** [review]
It's ok for me :+1:
